### PR TITLE
8266088: compiler/arguments/TestPrintOptoAssemblyLineNumbers test should user driver mode

### PR DESCRIPTION
**Pull request by iignatev**
Hi all,

could you please review this tiny patch for `TestPrintOptoAssemblyLineNumbers` test? 
from JBS:
> compiler/arguments/TestPrintOptoAssemblyLineNumbers class is a driver class that spawns a new JVM and validates its output, there is no need for the JVM which executes TestPrintOptoAssemblyLineNumbers to be run w/ all vm flags.

besides addressing this, the patch also removes `/compiler/patches` from `@library` as it's not needed and fixes code style here and there.

testing:
- [x] `compiler/arguments/TestPrintOptoAssemblyLineNumbers` on `{linux,windows,macosx}-x64-debug`

Thanks,
-- Igor

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266088](https://bugs.openjdk.java.net/browse/JDK-8266088): compiler/arguments/TestPrintOptoAssemblyLineNumbers test should user driver mode

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3726/head:pull/3726` \
`$ git checkout pull/3726`

Update a local copy of the PR: \
`$ git checkout pull/3726` \
`$ git pull https://git.openjdk.java.net/jdk pull/3726/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3726`

View PR using the GUI difftool: \
`$ git pr show -t 3726`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3726.diff">https://git.openjdk.java.net/jdk/pull/3726.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/compiler/arguments/TestPrintOptoAssemblyLineNumbers.java [hunk 1]
```diff
@@ -41,31 +41,31 @@
...
+        public static boolean foo(String arg) {
             return arg.contains("45");
         }
 
-        public static void main(String[] args){
+        public static void main(String[] args) {
             int count = 0;
-            for(int x = 0; x < 200_000; x++){
-                if(foo("something" + x)){ // <- test expects this line of code to be on line 68
+            for (int x = 0; x < 200_000; x++){
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/arguments/TestPrintOptoAssemblyLineNumbers.java:67]: Missed space between `){`

**Review by vnkozlov — CHANGES_REQUESTED** [review]
Add Oracle copyright line.

**Comment by iignatev** [comment]
staring more at this test, I've noticed a few other things which should be improved:
 - the test didn't use `OutputAnaluzer::*should*` method to check that the output contains the desired line. so in case of failure, we won't have all information about the subprocess
 - it didn't check the subprocess' exit code
 - the file is sensitive to line numbers but didn't have the comment which we use to mark and track such files

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by iignatev** [comment]
thanks, Vladimir.

/integrate

**Pull request closed without merging**